bpf/bpf_sock.c

```diff
@@ -1088,9 +1088,11 @@ __sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 	union v6addr addr6;
 
 	ctx_get_v6_address(ctx, &addr6);
+#ifdef ENABLE_IPV4
 	if (is_v4_in_v6(&addr6)) {
 		return __sock4_health_fwd(ctx);
 	} else {
+#endif /* ENABLE_IPV4 */
```

Comment:
Should the { go below here, so that when IPv4 is disabled the IPv6 code sits in the {} block?